### PR TITLE
Blogpost/metis

### PR DESCRIPTION
**Pull request by alanjschoen**
Please review this post, consider it a first draft.

### CODE REVIEW

File: _posts/2017-06-02-Metis_Project.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+keywords: data science, geospatial, machine learning, coding bootcamps, education
+published: True
+---
+
+Last year, I completed the Metis Data Science Bootcamp.  I had finished hiking the entire Appalachian Trail at the end of 2015, and then spent 6 grueling months looking for a job.  Once I had accumulated a few hundred rejection letters, and it seemed like I wasn't any closer to a job, I realized that my skills were outdated and I had to do something about it.  I looked into my options: coding bootcamps, graduate school, online graduate programs, and decided that a coding bootcamp was the fastest and most economical way to polish my skills.
+
+I had already worked in data science and computational modeling, but I had used Matlab as my main programming language for my entire professional career. At Metis, I learned about Python tools for data science, and I learned about how to present my work better.  About two weeks after graduating, I was fortunate to find my current job at DigitalGlobe.  Now, about 6 months later, I'm working with Metis to introduce more students to geospatial data and satellite imagery.
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: Second sentence.  How about "At Metis, I learned about Python tools for data science, and how to present my work better."

File: _posts/2017-06-02-Metis_Project.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+Metis students finish the program by completing a 3-week passion project.  The entire curriculum up until that point prepares students to take it on by developing skills like data acquisition, data wrangling, statistics, and model-building.  We partnered with Metis to provide data and tools for students who want to work with satellite images for their passion project.
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: You may want to specify who "we" is in the last sentence.

File: _posts/2017-06-02-Metis_Project.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+# Project Description
+We're exploring how to segment land types pixel-by-pixel in this project.  This will let us use machine learning for tasks like land-cover classification and land-use classification.  When we're getting value out of satellite images, this is one of the main ways to do it.
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: The last sentence doesn't really flow.  How about "These are two very important ways of getting value from satellite images."

File: _posts/2017-06-02-Metis_Project.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+Metis students finish the program by completing a 3-week passion project.  The entire curriculum up until that point prepares students to take it on by developing skills like data acquisition, data wrangling, statistics, and model-building.  We partnered with Metis to provide data and tools for students who want to work with satellite images for their passion project.
+
+# Project Description
+We're exploring how to segment land types pixel-by-pixel in this project.  This will let us use machine learning for tasks like land-cover classification and land-use classification.  When we're getting value out of satellite images, this is one of the main ways to do it.  
+There are many public sources of land cover data, like [GlobCover](http://due.esrin.esa.int/page_globcover.php), [NLCD](https://pubs.usgs.gov/fs/2012/3020/fs2012-3020.pdf), and [others](https://landcover.usgs.gov/uslandcover.php).  These are large projects published by government agencies.  They are great resources, but they lack the resolution we need for a lot of applications.
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: Last sentence, you can drop the second "they" and change "we need" to "needed"

File: _posts/2017-06-02-Metis_Project.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+When people need more accurate data, they often have to produce it themselves.  The Bill & Melinda Gates Foundation recently completed a census estimate in Nigeria using satellite imagery and machine learning ([described here](https://www.nature.com/news/satellite-images-reveal-gaps-in-global-population-data-1.21957)).  They sent out researchers to collect survey data, then they used that data to calibrate computer models to extend their estimates to more areas.  Our goal in this project is to use a similar methodology to train models to recognize objects and terrain that we're interested in.  Instead of collecting data door to door, we'll try to use publicly-available data from the internet.
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: Third sentence, you can drop the second "they"

File: _posts/2017-06-02-Metis_Project.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+![Population Estimate from the Gates Foundation (source: Nature Magazine)]({{ site.baseurl }}/assets/images/Metis_Project/nature-population-map-11-may-17-online.jpg){: width="40%"}
+
+
+In order to complete this project, we're using a machine learning modality called segmentation, which is related to the more commonly known task of classification, but differs in important ways.  Most people are familiar with image classification as an application of deep learning.  Classification is the challenge of assigning categories to pictures.  For example, a classifier could look at 1,000 pictures and tell you whether or not each one contains an airplane.  Segmentation goes a step further and outlines the plane in the image.
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: Split the first sentence.  I'd likely use "In order to complete this project, we're using a machine learning modality called segmentation.  Segmentation is related to the more commonly known task of classification, but differs in important ways."

Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: Last two sentences. How about "For example, a classifier looks at images and can tell you whether that image contains an object of interest, like an airplane.  Segmentation goes a step further and indicates *where* the airplane is in the image."

File: _posts/2017-06-02-Metis_Project.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+# Project Materials
+
+The main way I'm supporting this project is providing tutorials on [GitHub](https://github.com/DigitalGlobe/poirot).  I named the repository after a TV detective (perhaps the greatest TV detective) in keeping with the Metis convention for naming projects.  For the uninitiated, Hercule Poirot is basically the [Belgian James Bond](https://www.google.com/search?q=Hercule+Poirot&tbm=isch).
+
+My tutorials will show how to train a neural net to classify residential areas, forests, and water in Paris (yes, there are forests in Paris).
+
+## 0. Geospatial tools in python
+Working with geospatial data is a little different from working with other kinds of data.  Spatial is special, as they say.  So you need a special set of tools to work with geospatial data in Python, like GDAL, rasterio and GeoPandas.  These packages can be hard to set up and hard to use.  I provided a [couple](https://github.com/DigitalGlobe/poirot/blob/master/Setup%20(Mac).ipynb) of [notebooks](https://github.com/DigitalGlobe/poirot/blob/master/Setup%20(Ubuntu).ipynb) to help with the system setup.
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: I don't think the indication of the notebooks is clear enough.  I'd likely go with that sentence as is (minus links), and then add a [Mac], [Ubuntu] link at the end.

File: _posts/2017-06-02-Metis_Project.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+In order to train models, we'll need a collection of satellite images to use.  For that, we're turning to the SpaceNet dataset, a public dataset published by DigitalGlobe.  This is a collection of high-resolution imagery from 5 cities: Rio, Paris, Vegas, Khartoum, and Shanghai.  We originally published it as a resource for people coordinating the Rio Olympics.  We also used it to launch our first open data challenge on TopCoder in 2016.  Then we expanded the dataset to four more cities for the [second challenge](https://crowdsourcing.topcoder.com/spacenet) on TopCoder.  
+
+Most of the imagery is at 30cm resolution from our WorldView 3 satellite, except the Rio images, which are 50cm from WorldView 2.  In satellite imagery, resolution has a slightly different meaning than in most images.  The resolution refers to the size of each pixel, so in 30cm imagery each pixel is a 30cm X 30cm square.  There are, of course, some variations around that resolution depending on the [cartographic projection](https://xkcd.com/977/), but let's not complicate things too much.
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: Nice XKCD reference :+1:

File: _posts/2017-06-02-Metis_Project.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+We train models by showing them a series examples of images that we already marked up.  We could mark those images by hand, but that would be a lot of work.  It's much easier if we can find a source of data to use.  We call this ground truth.  We suggest that students use OpenStreetMap to get ground truth data.  OpenStreetMaps is a giant open database of map data including streets, buildings, and natural features.  At DigitalGlobe, we value OpenStreetMaps a lot, because its the most extensive public database of spatial data.  The [OpenStreetMaps Wiki](http://wiki.openstreetmap.org/wiki/Map_Features) shows you the sheer range of object categories that are in OpenStreetMaps.  Unfortunately, the data can be inaccurate and incomplete, especially for the rare categories that people don't use often, so you do need to make sure the data you want is good enough before you use it.
+
+You can get OSM data by downloading it from OSM, or you can get the same information packaged in files from [GEOFABRIK](http://www.geofabrik.de/).
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: link to OSM data?

File: _posts/2017-06-02-Metis_Project.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+## 4. Training a neural net
+[Notebook #4](https://github.com/DigitalGlobe/poirot/blob/master/4.%20Train%20a%20Model.ipynb) in the poirot repository shows how to train a U-Net segmentation neural network.
+
+Finally, after you have followed the first three notebooks, you're ready to train a neural net.  I used Keras and Tensorflow to train a U-Net network.  There are a lot of great segmentation networks out there, like [FCN](https://github.com/shelhamer/fcn.berkeleyvision.org), [Mask R-CNN](https://arxiv.org/abs/1703.06870), and [ResNet](https://github.com/DrSleep/tensorflow-deeplab-resnet), but I thought [U-Net](https://lmb.informatik.uni-freiburg.de/people/ronneber/u-net/) would be the best starting point for this project.  That's because U-Net is known for performing well when trained from scratch on small datasets.  That makes it a good starting point, because you can get decent performance without having to solve all the challenges of transfer learning
```
Comment (geovations) [inline _posts/2017-06-02-Metis_Project.md]: I'd move the last sentence up to be the third sentence in the second paragraph.  In addition, there is no punctuation after "transfer learning."

File: _posts/2017-06-02-Metis_Project.md (hunk 9 above)
Comment (alanjschoen) [inline _posts/2017-06-02-Metis_Project.md]: One of the notebooks is about how to download OSM data directly into python

File: _posts/2017-06-02-Metis_Project.md (hunk 8 above)
Comment (alanjschoen) [inline _posts/2017-06-02-Metis_Project.md]: That comic is in 2/4 of my blog posts

**Pull request merged**